gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/filter/AbstractJWTFilter.java

```diff
@@ -370,6 +375,11 @@ protected boolean validateToken(final HttpServletRequest request, final HttpServ
     return false;
   }
 
+  private boolean isTokenEnabled(String tokenId) throws UnknownTokenException {
+    final TokenMetadata tokenMetadata = tokenStateService == null ? null : tokenStateService.getTokenMetadata(tokenId);
```

Comment:
Under what circumstances we don't have metadata for a token? Is it a valid scenario or should it only happen by accident?

Reply:
There are 2 possible scenarios where there is no token metadata:

1. if the token management is turned off for the given topology (on KNOXTOKEN service)
2. the metadata alias is something that we invented in a later phase of Knox token management (but before adding the JDBC implementation). So in theory it may happen that only the basic information is stored in the underlying token store (keystore, Zookeeper) but the metadata is not yet there.

In both cases, we should consider the given token as an enabled one.

Reply:
Ok, makes sense.